SECURITY.md

```diff
@@ -256,6 +256,14 @@ embargo policy to share binary distributions of the security fixes before the pu
 This includes, but is not limited to, Envoy binaries and Docker images. It is expected that
 distributors have a method to stage and validate new binaries without exposing them publicly.
 
+If Envoy is running in a hosted environment, the fix may rollout ahead of the embargo lift date
+only if the end user **cannot** access the running Envoy process and possibly reverse engineer the fix.
```

Comment:
"and possibly reverse engineer the fix" leaves room for a lot of ambiguity and an open ended clause IMO. I would spell out defined clauses and add a final decision making statement like this:

1.  If the end user **cannot** access the running Envoy process
2. If the end user **cannot** derive the Envoy version from responses or any change in the response behavior

The Envoy Security Team will decide the eligibility of each CVE on a case by case basis and convey to the vendors if a patch can be deployed before embargo is lifted.

Reply:
IIUC if the fix in Envoy, users will mostly likely see change in response behavior otherwise you won't be fixing it, so I don't get the whole deploy before lifting embargo argument.

Reply:
I don't think deriving the Envoy version is particularly useful from a security perspective. The key point is whether they can use environmental signals to reverse engineer the CVE and the fix. At that point I agree that it's at the discretion of the EST and there should be guidance for each fix in a release.